packages/interfaces/src/filter.ts

```diff
@@ -1,18 +1,23 @@
+import { EventEmitter } from "@libp2p/interface/events";
 import type { PeerId } from "@libp2p/interface/peer-id";
 
 import type { IDecodedMessage, IDecoder } from "./message.js";
 import type { ContentTopic } from "./misc.js";
-import type { Callback, IBaseProtocol } from "./protocols.js";
+import type { IBaseProtocol } from "./protocols.js";
 import type { IReceiver } from "./receiver.js";
 
 export type ContentFilter = {
   contentTopic: string;
 };
 
-export interface IFilterSubscription {
+export interface SubscriptionEventMap {
+  [key: string]: CustomEvent<IDecodedMessage>;
```

Comment:
As I proposed https://github.com/waku-org/js-waku/issues/1683#issuecomment-1796456857 we should make it possible for client to react to errors.

But looking at this PR I think we cannot easily match particular content topic and error.
Instead let's do generic `error` event to which client can subscribe and react to.